### ISSUE TITLE
backup-list --detail fails on missing metadata.json

### ISSUE THREAD

**Issue by neale-bpm** [wal-g/wal-g#556]
I'm working on a script to create a standby cluster based on how Patroni's patroni_wale_restore worked.
However, it seems wal-g added support for --detail, but this fails as follows on our existing backups (and naturally support to specify LATEST as an arg would be nice ;-) )

```
# envdir /data/conf/wal-g.env wal-g backup-list --detail
ERROR: 2020/02/07 19:11:47.445792 object '/backup-disk/basebackups_005/base_000000020000000000000006/metadata.json' not found in storage
```

**Comment by g0djan** [comment]
Hi, we can't remove using metadata.json from backup's listing because there is honest information about times of start and finish of backup. If you don't need this information in output possibly we can add a flag --without-times or something like this.

**Comment by x4m** [comment]
Hi! Actually, the problem is that backups are taken with old version of wal-g that did not place metadata

Actually, wal-g should be compatible with it's own versions within one major release. So, if --detail existed before metadata, we should condier this as a bug

**Comment by neale-bpm** [comment]
Yes. It does appear to be a bug that --detail breaks, especially as there may be backups in that list that do have the metadata (e.g. the most recent ones since an upgrade of wal-g).

**Comment by Andrey-Salnikov** [comment]
I've met the same bug, and this message appears even a backup has been made with latest version of wal-g. I have history of backups made by different versions of wal-g including latest version v0.2.15.

**Comment by pawanrawal** [comment]
We also face this issue with our cluster. We have an auto-expiration policy for older backups on S3. Sometimes the `metadata.json` file is deleted before the backup folder itself and then our automated pipelines breaks because `wal-g backup-list --detail` fails. Are there any chances of this getting fixed @x4m?

**Comment by usernamedt** [comment]
> We also face this issue with our cluster. We have an auto-expiration policy for older backups on S3. Sometimes the `metadata.json` file is deleted before the backup folder itself and then our automated pipelines breaks because `wal-g backup-list --detail` fails. Are there any chances of this getting fixed @x4m?

I think it is a different problem. It appears to be that the initial problem is related to compatibility issues because of `metadata.json` introduction in the newer WAL-G versions.

As for now, WAL-G expects `metadata.json`, `*_backup_stop_sentinel.json`, and `files_metadata.json` files to exist in storage.
